src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/UriPresentation/RemoteUriPresentationService.cs

```diff
@@ -37,15 +37,15 @@ internal sealed class RemoteUriPresentationService(
 
     private async ValueTask<TextChange?> GetPresentationAsync(Solution solution, DocumentId razorDocumentId, LinePositionSpan span, Uri[]? uris, CancellationToken cancellationToken)
     {
-        var razorDocument = solution.GetAdditionalDocument(razorDocumentId);
-        if (razorDocument is null)
+        if (await GetRazorCodeDocumentAsync(solution, razorDocumentId).ConfigureAwait(false) is not { } codeDocument)
         {
             return null;
         }
 
-        var snapshot = _documentSnapshotFactory.GetOrCreate(razorDocument);
-        var codeDocument = await snapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
-        var sourceText = await razorDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+        var razorDocument = solution.GetAdditionalDocument(razorDocumentId);
```

Comment:
I see what you're going for with the base class, and I like the initiative, but this line irks me, because we've already done this work in the base class. I think we need to think about the design of this a bit more in a follow up at some point.

Reply:
Yeah, I was thinking of maybe having GetCodeAndRazorDocuments and then a wrapper that discards the one we don't need. Does that sound reasonable?